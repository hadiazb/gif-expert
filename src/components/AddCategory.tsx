import { useState } from 'react';

export interface AddCategoryProps {
	onNewCategory: (category: string) => void;
}

export const AddCategory: React.FC<AddCategoryProps> = ({ onNewCategory }) => {
	const [inputValue, setInputValue] = useState<string>('');

	const inputOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
		setInputValue(event.target.value);
	};

	const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
		event.preventDefault();
		if (inputValue.trim().length > 0) {
			onNewCategory(inputValue.trim());
			setInputValue('');
		}
	};

	return (
		<form onSubmit={onSubmit}>
			<input
				type='text'
				placeholder='Buscar gifs'
				value={inputValue}
				onChange={inputOnChange}
			/>
		</form>
	);
};
