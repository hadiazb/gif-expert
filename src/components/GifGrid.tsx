import { GifItem } from '.';
import { useFetchGifs } from '../hooks';

export interface GifGridProps {
	category: string;
}

export const GifGrid: React.FC<GifGridProps> = ({ category }) => {
	const { images, isLoading, error } = useFetchGifs(category);

	return (
		<div>
			{isLoading && <p>Cargando...</p>}
			<>
				{error ? (
					<p>{error}</p>
				) : (
					<>
						<h3>{category}</h3>
						<div className='card-grid'>
							{images.map((image) => (
								<GifItem key={image.id} {...image} />
							))}
						</div>
					</>
				)}
			</>
		</div>
	);
};
