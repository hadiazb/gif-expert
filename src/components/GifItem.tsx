interface GifItemProps {
	title: string;
	url: string;
}

export const GifItem: React.FC<GifItemProps> = ({ title, url }) => {
	return (
		<div className='card'>
			<img src={url} alt={title} />
			<p>{title || 'sin titulo'}</p>
		</div>
	);
};
