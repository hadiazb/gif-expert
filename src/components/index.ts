export * from './GifItem';
export * from './gifSearch.interface';
export * from './GifGrid';
export * from './AddCategory';
