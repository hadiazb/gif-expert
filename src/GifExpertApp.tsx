import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
	const [categories, setCategories] = useState<string[]>([]);

	const onAddCategory = (newCategory: string) => {
		if (categories.includes(newCategory)) return;
		setCategories((categories) => [newCategory, ...categories]);
	};

	return (
		<>
			<h1>GifExpertApp</h1>
			<AddCategory onNewCategory={(event) => onAddCategory(event)} />
			<ol>
				{categories.length > 0 &&
					categories.map((category) => (
						<GifGrid key={category} category={category} />
					))}
			</ol>
		</>
	);
};
