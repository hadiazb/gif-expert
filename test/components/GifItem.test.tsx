import { render, screen } from '@testing-library/react';

import { GifItem } from '../../src/components';

describe('GifItem', () => {
	const title = 'titulo';
	const url = 'https://one-punch.com/saitama.jpg';

	test('Snapshot the component', () => {
		const { container } = render(<GifItem title={title} url={url} />);
		expect(container).toMatchSnapshot();
	});

	test('should the image with url and alt tags ', () => {
		render(<GifItem title={title} url={url} />);

		const { src, alt } = screen.getByRole('img') as HTMLImageElement;

		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('should the title in the component', () => {
		render(<GifItem title={title} url={url} />);

		expect(screen.getByText(title)).toBeTruthy();
	});
});
