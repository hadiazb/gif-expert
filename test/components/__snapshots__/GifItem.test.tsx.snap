// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Snapshot the component 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
